mlrun/model_monitoring/db/tsdb/base.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from abc import ABC
+
+import pandas as pd
+
+
+class TSDBConnector(ABC):
+    def __init__(self, project: str):
+        """
+        Initialize a new TSDB connector.
+        :param project: the name of the project.
+        """
+        self.project = project
+
+    def apply_monitoring_stream_steps(self, graph):
+        """
+        Apply TSDB steps on the provided monitoring graph. Throughout these steps, the graph stores live data of
+        different key metric dictionaries. This data is being used by the monitoring dashboards in
+        grafana.
+        There are 3 different key metric dictionaries that are being generated throughout these steps:
+        - base_metrics (average latency and predictions over time)
+        - endpoint_features (Prediction and feature names and values)
+        - custom_metrics (user-defined metrics)
+        """
+        pass
+
+    def write_application_event(self, event: dict):
+        """
+        Write a single application result event to TSDB.
+        """
+        pass
+
+    def delete_tsdb_resources(self):
+        """
+        Delete all project resources in the TSDB connector, such as model endpoints data and drift results.
+        """
+        pass
+
+    def get_model_endpoint_real_time_metrics(
```

Comment:
Since some of the functionality might be relatet to V3IO (such as how we process the df that is returned from get_records()), i kept it in the v3io connector. I did remove the relevant func from the kv_store and moved it into the crud part. 
Anyway, if we will see that this behaviour is repeating in other TSDBs (TDEngine) then I will move all of the logic to the crud part. Let me know if it's not clear